### PR TITLE
TRY: github actions for automated tests instead of travis

### PR DESCRIPTION
**Pull request by aaronjorbin**

### CODE REVIEW

**Comment by aaronjorbin** [comment]
@desrosj Would you be willing to give this a review?

**Comment by aaronjorbin** [comment]
Waiting for #98 so that I can turn off travis but still get automated test runs

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Npm install and build
```
Comment (desrosj) [inline .github/workflows/test.yml]:
```suggestion
    - name: Install NPM dependencies and build
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      run: |
+        npm ci
+
+    - name: Running the tests
```
Comment (desrosj) [inline .github/workflows/test.yml]:
```suggestion
    - name: Run tests
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    name: JavaScript
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-versions: [12.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (desrosj) [inline .github/workflows/test.yml]:
```suggestion
    - name: Checkout repository
      uses: actions/checkout@v2
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Npm install and build
+      run: |
+        npm ci
```
Comment (desrosj) [inline .github/workflows/test.yml]:
```suggestion
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Npm install and build
+      run: |
```
Comment (desrosj) [inline .github/workflows/test.yml]:
```suggestion
      run: npm ci
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  pull_request:
+    paths-ignore:
+    - '**.md'
+  push:
+    branches: [master]
+    paths-ignore:
+    - '**.md'
+
+jobs:
+  test:
```
Comment (desrosj) [inline .github/workflows/test.yml]:
```suggestion
  test-js:
```

Comment (desrosj) [reply .github/workflows/test.yml]: Just in case we add other types of testing later.

File: .github/workflows/test.yml (hunk 3 above)
Comment (desrosj) [reply .github/workflows/test.yml]: Having a name for each step makes it more clear what's happening.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    name: JavaScript
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-versions: [12.x, 14.x]
```
Comment (desrosj) [inline .github/workflows/test.yml]:
```suggestion
        node-versions: [ '12', '14' ]
```

Comment (desrosj) [reply .github/workflows/test.yml]: 12.x will work, but `12` will do the same.

File: README.md [hunk 8]
```diff
@@ -1,7 +1,5 @@
 # grunt-patch-wordpress
```
Comment (desrosj) [inline README.md:3]:
```suggestion
![Unit Tests](https://github.com/WordPress/grunt-patch-wordpress/workflows/Unit%20Tests/badge.svg)

```

**Review by desrosj — COMMENTED** [review]
Looks good! I made some small suggestions.

I do like to include inline docs for each job summarizing all steps that will happen (see [some of the ones in Core for an example](https://core.trac.wordpress.org/browser/trunk/.github/workflows/php-compatibility.yml#L14)), but not required.

**Pull request merged**